src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/SparkSessionTests.cs

```diff
@@ -65,5 +68,109 @@ public void TestSignaturesV2_4_X()
+            }
+
+            // Calling CreateDataFrame(IEnumerable<int> _) without schema
+            {
+                var data = new List<int>(new int[] { 1, 2 });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new IntegerType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<double> _) without schema
+            {
+                var data = new List<double>(new double[] { 1.2, 2.3 });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new DoubleType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<bool> _) without schema
+            {
+                var data = new List<bool>(new bool[] { true, false });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new BooleanType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<long> _) without schema
+            {
+                var data = new List<long>(new long[] { 2L, 3L });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new LongType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+        }
+
+        /// <summary>
+        /// Validates that the dataframe returned by the CreateDataFrame API is correct as per the
+        /// given data.
+        /// </summary>
+        /// <param name="data">Given data for the dataframe</param>
+        /// <param name="df">Dataframe Object to validate</param>
+        /// <param name="schema">Expected schema of the dataframe</param>
+        private void ValidateDataFrame(IEnumerable<object[]> data, DataFrame df, StructType schema)
+        {
+            Assert.Equal(schema, df.Schema());
+            Row[] rows = df.Collect().ToArray();
+            Assert.Equal(data.Count(), rows.Length);
+            int i = 0;
+            foreach (object[] expectedValues in data)
+            {
+                object[] actualValues = rows[i].Values;
+                Assert.Equal(expectedValues.Length, actualValues.Length);
+                Assert.True(actualValues.SequenceEqual(expectedValues));
```

Comment:
Let's consolidate these three lines to:
```csharp
Assert.Equal(expectedValues, rows[i].Values);
```
xUnit will take the length into account as well for the comparison.